MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt

```diff
@@ -0,0 +1,252 @@
+package com.mapbox.vision.sync
+
+import android.app.Application
+import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.vision.BuildConfig
+import com.mapbox.vision.mobile.core.NativeVisionManager
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.models.VideoClip
+import com.mapbox.vision.mobile.core.telemetry.TelemetryImageSaver
+import com.mapbox.vision.mobile.core.utils.extentions.TAG_CLASS
+import com.mapbox.vision.sync.metagenerator.TelemetryMetaGenerator
+import com.mapbox.vision.sync.session.RecordingSessionWriter
+import com.mapbox.vision.sync.session.RotatedBuffersSessionWriter
+import com.mapbox.vision.sync.session.SessionWriter
+import com.mapbox.vision.sync.syncmanager.TelemetrySyncManager
+import com.mapbox.vision.sync.telemetry.TelemetryImageSaverImpl
+import com.mapbox.vision.utils.FileUtils
+import com.mapbox.vision.utils.VisionLogger
+import com.mapbox.vision.utils.file.RotatedBuffers
+import com.mapbox.vision.video.videoprocessor.VideoProcessor
+import com.mapbox.vision.video.videoprocessor.VideoProcessorListener
+import com.mapbox.vision.video.videosource.camera.VideoRecorder
+import java.io.File
+
+internal interface SessionManager {
+
+    fun start()
+
+    fun stop()
+
+    fun setCountry(country: Country)
+
+    fun startRecording(path: String)
+
+    fun stopRecording()
+
+    class Impl(
+        private val application: Application,
+        private val nativeVisionManager: NativeVisionManager,
+        private val videoRecorder: VideoRecorder,
+        private val mapboxTelemetry: MapboxTelemetry,
+        private val telemetryImageSaver: TelemetryImageSaver
+    ) : SessionManager {
+
+        companion object {
+            private const val CACHE_DIR = "Cache"
+            private const val VIDEO_BUFFERS_DIR = "Buffers"
+            private const val DIR_TELEMETRY = "Telemetry"
```

Comment:
rename to TELEMETRY_DIR to be consistent

Reply:
refactor